libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -563,6 +573,7 @@ void HTTPClient::setRedirectLimit(uint16_t limit)
 void HTTPClient::useHTTP10(bool useHTTP10)
 {
     _useHTTP10 = useHTTP10;
+    _reuse = !useHTTP10;
```

Comment:
With this method and `::setReuse()` it is possible to set `useHTTP10 = true` and `_reuse=true`.  That is the user shooting themselves in the foot, which is not a high prio for me.

At some point might want to go back and see if `useHTTP10 === !reuse` (I think so after reading the code briefly) and then drop one or the other variable if that's true.  But for this PR, I'm OK.

Reply:
@earlephilhower for HTTP/1.0 the default is *not* reusing the connection, the default for HTTP/1.1 is reusing. However reusing is allowed also on HTTP/1.0. In this case the client must send a Connection: Keep-Alive request header. The server may respond with a Connection: Keep-Alive in which case the connection can be reused. In HTTP/1.1 the client may send a connection: Keep-Alive header, but this is not obligatory. The connection is supposed to be reused unless the *server* sends a connection: close response header.
This is exactly what ESP8266HTTPClient does